### PR TITLE
Build kopia binary as part of kanister-tools

### PR DESCRIPTION
**Pull request by ankitjain235**
## Change Overview

This PR updates workflow to build Kopia binary as a part of Kanister-tools image build instead of a separate image.

## Pull request type

Please check the type of change your PR introduces:
- [ ] :construction: Work in Progress
- [ ] :rainbow: Refactoring (no functional changes, no api changes)
- [ ] :hamster: Trivial/Minor
- [ ] :bug: Bugfix
- [ ] :sunflower: Feature
- [ ] :world_map: Documentation
- [ ] :robot: Test

## Issues 

- fixes #issue-number

## Test Plan

- [ ] :muscle: Manual
- [ ] :zap: Unit test
- [ ] :green_heart: E2E

### CODE REVIEW

File: docker/tools/Dockerfile [hunk 1]
```diff
@@ -1,3 +1,49 @@
+# Build Kopia binary
+FROM golang:1.19-bullseye AS builder
+
+ARG kopiaBuildCommit
+ARG kopiaRepoOrg
```
Comment (PrasadG193) [inline docker/tools/Dockerfile:5]: I think best practice is to use `snake_case` instead of camelCase for ARG and ENV

Comment (PrasadG193) [reply docker/tools/Dockerfile:5]: If arg is not set, does it fail the build?

Comment (ankitjain235) [reply docker/tools/Dockerfile:5]: We do expect the arg to be always set.

Comment (julio-lopez) [reply docker/tools/Dockerfile:5]: Ankit: it is expected to be set when called via `goreleaser`. What happens if someone tries to build the image directly? what happens if the argument is unset? do things fails?

**Comment by julio-lopez** [comment]
This is the right direction, there is no need for building kopia in a separate image and pushing it to a registry.

File: docker/tools/Dockerfile (hunk 1 above)
Comment (ankitjain235) [reply docker/tools/Dockerfile]: I have kept default values in Dockerfile, allowing them to be passed using build-arg if required.

File: docker/tools/Dockerfile [hunk 2]
```diff
@@ -1,3 +1,50 @@
+# Build Kopia binary
+FROM golang:1.19-bullseye AS builder
+
+# Refers to https://github.com/kopia/kopia/commit/317cc36892707ab9bdc5f6e4dea567d1e638a070
+ARG kopiaBuildCommit=317cc36
```
Comment (PrasadG193) [inline docker/tools/Dockerfile:5]: Should we use the latest commit by default? Maybe a master branch?

Comment (ankitjain235) [reply docker/tools/Dockerfile:5]: Since we are cloning, it would be latest commit always so we can leave it empty as well. Setting master might confuse others if it should be branch instead of commit.

Comment (PrasadG193) [reply docker/tools/Dockerfile:5]: sounds good

File: docker/tools/Dockerfile [hunk 3]
```diff
@@ -1,3 +1,46 @@
...
+
+# Add CA certs
+RUN apt-get update && apt-get -y install ca-certificates && \
+  rm -rf /var/cache/apk/* && \
+  adduser kopia && addgroup kopia kopia && \
+  chown kopia /kopia
+
+USER kopia:kopia
+
+ENTRYPOINT [ "/kopia/kopia" ]
```
Comment (pavannd1) [inline docker/tools/Dockerfile:41]: Why is this required?

**Review by pavannd1 — APPROVED** [review]
LGTM
